### PR TITLE
[KOGITO-8805] - [KSW-Operator] Workflow Dev Profile Builder Image lac…

### PR DESCRIPTION
**Pull request by spolti**
…king permissions to write on mvn dir on OpenShift

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO|RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [x] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- <b>(Re)run Jenkins tests</b>  
  Please add comment: <b>Jenkins [test|retest] this</b>

- <b>Prod tests</b>  
  Please add comment: <b>Jenkins (re)run [prod|Prod|PROD]</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by spolti** [comment]
jenkins test this

**Comment by spolti** [comment]
jenkins retest this

**Comment by kie-ci1** [comment]
**PR job** `#571` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/571/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Prepare offline kogito-examples)<br/>Stage "Prepare offline kogito-examples" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build & Test Images)<br/>Stage "Build & Test Images" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/workspace/KIE/kogito/main/pullrequest/kogito-images/workspaces<br/>[Pipeline] sh<br/>++ docker ps -a -q<br/>+ docker rm -f<br/>"docker rm" requires at least 1 argument.<br/>See 'docker rm --help'.<br/><br/>Usage:  docker rm [OPTIONS] CONTAINER [CONTAINER...]<br/><br/>Remove one or more containers<br/>+ date<br/>Tue Mar 28 06:19:06 EDT 2023<br/>[Pipeline] sh<br/>++ docker images -q<br/>+ docker rmi -f<br/>"docker rmi" requires at least 1 argument.<br/>See 'docker rmi --help'.<br/><br/>Usage:  docker rmi [OPTIONS] IMAGE [IMAGE...]<br/><br/>Remove one or more images<br/>+ date<br/>Tue Mar 28 06:19:07 EDT 2023<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.cekit/cache<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/571/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by spolti** [comment]
@radtriste ideas about why it is failing?

**Comment by spolti** [comment]
Jenkins run PROD

**Comment by kie-ci1** [comment]
**PR job** `#573` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/573/display/redirect

**Test results:**
- PASSED: 257
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/573/testReport/features.image.kogito-swf-builder/SWF and Quarkus installation/verify_if_container_starts_in_devmode_by_default/">features.image.kogito-swf-builder.SWF and Quarkus installation.verify if container starts in devmode by default</a></summary>
Message 'MAVEN_ARGS_APPEND= -X --batch-mode -Dmaven.compiler.release=11 --no-transfer-progress -Duser.home=/home/kogito' was not found in the logs
</details>

**Comment by spolti** [comment]
Jenkins run PROD

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
@davidesalerno if you are ok with that, we can merge it :)

File: modules/kogito-maven/3.8.x/added/configure-maven.sh [hunk 1]
```diff
@@ -41,6 +42,13 @@ function configure() {
     rm -rf *.bak
 }
 
+# When Running on OpenShift with AnyUID the HOME environment variable gets overridden to "/"
+# Maven build main fail with this issue Could not create local repository at /.m2/repository
```
Comment (spolti) [inline modules/kogito-maven/3.8.x/added/configure-maven.sh]:
```suggestion
# Maven build might fail with this issue:  'Could not create local repository at /.m2/repository'
```

**Comment by spolti** [comment]
Jenkins run PROD

File: modules/kogito-maven/3.8.x-rpm/added/configure-maven.sh [hunk 2]
```diff
@@ -79,7 +90,7 @@ function configure_proxy() {
         fi
         xml="$xml\
        </proxy>"
-        sed -i "s|<!-- ### configured http proxy ### -->|${xml}|" "${MAVEN_SETTINGS_PATH}"
+        sed -i.bak "s|<!-- ### configured http proxy ### -->|${xml}|" "${MAVEN_SETTINGS_PATH}"
```
Comment (davidesalerno) [inline modules/kogito-maven/3.8.x-rpm/added/configure-maven.sh:93]: What is it the purpose of adding `.bak `?

**Review by davidesalerno — APPROVED** [review]

**Pull request merged**